### ISSUE TITLE
Latest Issues

### ISSUE THREAD

**Issue by DominusMaximus** [demilich1/metastone#293]
- Dispatch Kodo's battlecry damage should be increased by auras such as Leokk's, or by playing it adjacent to a Dire Wolf Alpha.

- With the recent hotfix, Daring Reporter should no longer trigger if it's mortally wounded (it can no longer survive when mortally wounded by a Mortal Coil). Here are some more details: http://hearthstone.gamepedia.com/Daring_Reporter

- Despite the wording on King's Elekk, a successful joust isn't considered a draw. This means it shouldn't trigger cards such as Daring Reporter or Chromaggus. 

- According to this http://hearthstone.gamepedia.com/Excess_Mana if you play an Innervate while you have 8/8 mana crystals, and then play a Wild Growth that's been reduced to 0 mana, you should receive an Excess Mana card.

- Hidden Cache shouldn't trigger if you have no minions in hand. 

- Why did you revert Doppelgangster's spelling? Your spelling was correct!

- If I play a 1 Attack minion with a Hobgoblin and Leokk on board, Hobgoblin should trigger before Leokk's aura affects the minion.

- If I play a Twilight Drake with a Steward of Darkshire on board, the Drake's battlecry should resolve after Steward of Darkshire triggers, resulting in a Twilight Drake with Divine Shield.

- If I have a Hobgoblin on board and play a Faceless Shambler (targeting a 1/1 minion), the Shambler's stats should receive the Hobgoblin buff, but then be overwritten by its own battlecry, resulting in a 1/1 minion. In Metastone it ends up as a 3/3.

**Comment by webadict** [comment]
1. This is odd... I thought this already worked? It's weird that Mirror Entity updates to reflect the Aura, but the Battlecry doesn't.

2. Ezpz. And by that I mean crap. Let's see if I can hack the problem away.

3. Yeah, I suppose that one's a fairly easy fix. The problem is that you need to remove the card from the deck and then add it to your hand, which is super lame, considering. Luckily, this is fixable.

4. WTF? How is that even a thing? I guess the condition should be if (Max Mana | Current Mana) == 10

5. Yes, good catch.

6. I get yelled at for spelling it wrong... I get yelled at for spelling it right... I get yelled at for spelling it wrong again... Will there be no end to the yelling?

**Comment by DominusMaximus** [comment]
**1.** My guess is that in Metastone, Battlecries are currently resolving before Auras are applied, and Mirror Entity is resolving after both. I couldn't come up with a way to test this, but here are some other things I just found: 

- When I was testing Mirror Entity interactions for Small-Time Buccaneer, I tested Hobgoblin with Mirror Entity, but didn't test Hobgoblin with Mirror Entity _and_ a Leokk on board. It turns out the Hobgoblin won't trigger if there's a Leokk on board.

- A similar interaction that isn't working is Steward of Darkshire + Twilight Drake. The battlecry should resolve after Steward of Darkshire triggers, resulting in a Twilight Drake with Divine Shield. 

- Here's another one: Hobgoblin + Faceless Shambler. If you were to target a 1/1 minion with Faceless Shambler, the Shambler's stats should receive the Hobgoblin buff, but then be overwritten by its own battlecry, resulting in a 1/1 minion. In Metastone it ends up as a 3/3. 

I added those 3 to the original post. 

**2.** I find it silly how Daring Reporter will still survive the Wrath version with draw, but not the Fandral version.

**6.** It appears our Battlecries, Auras and Triggers were resolving with incorrect timings, resulting in this miscommunication!

**Comment by webadict** [comment]
1. There is currently an issue where the board isn't updated to reflect Auras just yet. This means that only minions with Battlecries would have their stats updated after it procs. Putting in an updated board state beforehand seems to fix this.
- As for these additional bugs, this appears to be several trigger timing issues. So, I guess I have to add events, so that triggers will happen correctly. We're already missing an AfterMinionSummoned/Played Event, so I'll add a BeforeMinionSummoned/Played Event as well? Hearthstone, wtf?
2. It's a bug with how the card is programmed on Hearthstone's side. It absolutely should kill it, but because Hearthstone lives in lala land, and nobody cares about consistency, it will probably be like that forever. However, I've updated the card to work accordingly (I also had to "fix" Wrath...)
6. I blame Brode.
